corehq/apps/linked_domain/views.py

```diff
@@ -186,3 +192,114 @@ def delete_domain_link(self, in_data):
         return {
             'success': True,
         }
+
+
+class DomainLinkHistoryReport(GenericTabularReport):
+    name = 'Project Link History'
+    base_template = "reports/base_template.html"
+    section_name = 'Project Settings'
+    slug = 'project_link_report'
+    dispatcher = DomainReportDispatcher
+    ajax_pagination = True
+    asynchronous = False
+    sortable = False
+
+    @property
+    def fields(self):
+        if self.master_link:
+            fields = []
+        else:
+            fields = ['corehq.apps.linked_domain.filters.DomainLinkFilter']
+        fields.append('corehq.apps.linked_domain.filters.DomainLinkModelFilter')
+        return fields
+
+    @property
+    def link_model(self):
+        return self.request.GET.get('domain_link_model')
+
+    @property
+    @memoized
+    def domain_link(self):
+        if self.request.GET.get('domain_link'):
+            try:
+                return DomainLink.objects.get(pk=self.request.GET.get('domain_link'))
+            except DomainLink.DoesNotExist:
+                pass
```

Comment:
Should this 404?

Reply:
Raising a 404 here would 404 the whole report. There is a check in `get_data` that returns no rows if the link doesn't exist.